indexer/src/main.ts

```diff
@@ -73,206 +79,238 @@ export default async function transform({
-      if (!isKakarotTx) {
-        return;
-      }
-
-      const ethTx = toEthTx({
-        transaction,
-        receipt,
-        blockNumber,
-        blockHash,
-        isPendingBlock,
-      });
-      if (!ethTx) {
-        return;
-      }
-
-      // Get the cumulative gas used for the reverted transaction.
-      // Example:
-      // const cumulativeGasUsages = [300n, undefined, undefined, 200n, undefined, 100n, undefined, undefined, 10n, undefined];
-      // const ethTx = { transactionIndex: 5 };
-      // const revertedTransactionCumulativeGasUsed = 100n;
-      const len = cumulativeGasUsages.length;
-      const revertedTransactionCumulativeGasUsed =
-        cumulativeGasUsages.find((gas, i) => {
-          return (
-            Number(ethTx.transactionIndex) >= len - 1 - i && gas !== undefined
-          );
-        }) ?? 0n;
-
-      const ethReceipt = toRevertedOutOfResourcesReceipt({
-        transaction: ethTx,
-        blockNumber,
-        blockHash,
-        cumulativeGasUsed: revertedTransactionCumulativeGasUsed,
-        isPendingBlock,
-      });
-
-      store.push({ collection: "transactions", data: { tx: ethTx } });
-      store.push({
-        collection: "receipts",
-        data: {
-          receipt: ethReceipt,
-        },
-      });
-    }
-  });
+  const processedTransactions = processTransactions(
+    transactions,
+    blockInfo,
+    cumulativeGasUsages,
+  );
+  updateStoreWithTransactions(store, processedTransactions);
 
   const ethHeader = await toEthHeader({
     header: header,
     gasUsed: cumulativeGasUsed,
     logsBloom: blockLogsBloom,
     receiptRoot: receiptTrie.root(),
     transactionRoot: transactionTrie.root(),
-    blockNumber,
-    blockHash,
-    isPendingBlock,
+    ...blockInfo,
   });
   store.push({
-    collection: "headers",
+    collection: Collection.Headers,
     data: { header: ethHeader },
   });
 
   return store;
 }
+
+function createBlockInfo(header: BlockHeader): BlockInfo {
+  const blockNumber = padString(toHexString(header.blockNumber), 8);
+  const blockHash = padString(header.blockHash, 32);
+  const isPendingBlock = blockHash === NULL_BLOCK_HASH;
+  return { blockNumber, blockHash, isPendingBlock };
+}
+
+function processEvent(blockInfo: BlockInfo) {
+  return (event: EventWithTransaction): ProcessedEvent | null => {
+    const typedEthTx = toTypedEthTx({ transaction: event.transaction });
+    // Can be null if:
+    // 1. The transaction is missing calldata.
+    // 2. The transaction is a multi-call.
+    // 3. The length of the signature array is different from 5.
+    // 4. The chain id is not encoded in the v param of the signature for a
+    //    Legacy transaction.
+    // 5. The deserialization of the transaction fails.
+    if (!typedEthTx) return null;
+
+    const ethTx = typedTransactionToEthTx({
+      typedTransaction: typedEthTx!,
+      receipt: event.receipt,
+      ...blockInfo,
+    });
```

Comment:
there is a naming mistmatch here. We first create a "eth" tx, then say that it's a "typed" tx (which also looks like a eth tx to me) so that eventually it becomes a "eth" tx again

Reply:
The function toTypedEthTx converts the transaction into a typed Ethereum transaction, returning an object of type TypedTransaction. 
On the other hand, typedTransactionToEthTx takes this TypedTransaction and converts it into the Ethereum transaction format, returning an object of type ExtendedJsonRpcTx.

In both cases, they are eth transactions, but they return objects of different types, which is why those names were chosen. What do you think?